### ISSUE TITLE
Externalise the configuration of WellKnownFeaturePacks

### ISSUE THREAD

**Issue by jmesnil** [wildfly-extras/prospero#93]
All enum values in `WellKnownFeaturePacks` should be derived from an external config file.

This will allow to make sure that Prospero either deal with EAP or WildFly but not both. 

One way to externalise these values is with a properties file that list all required settings, with something like:

```
# list the keys of all known feature pack
prospero.feature-packs= x, y, z
# each feature pack defines its own properties
prospero.feature-pack.x.location=...
prospero.feature-pack.x.channel=...
# optional
prospero.feature-pack.x.pakages=...
# optional
prospero.feature-pack.x.repositories=...
```

**Comment by jmesnil** [comment]
The values of the feature packs must also be refined:

* WildFly does not provide channels at `org.wildfly.channels:wildfly`
* There will be no channels for EAP 7.4
* eap-8.0 should be renamed to eap-8.0-beta
  * it will be available in MRRC EA, not MRRC GA
* `eap` should be removed as we are not providing channels across EAP minor release 
* `xp-5.0` should be renamed to `eap-xp-5` 
* `xp` should be removed as we are not providing channels across XP major release

**Comment by spyrkob** [comment]
`eap` and `xp` were meant as shortcuts for latest known micro rather then subscribing to minor/major releases. 
So if there `eap-8.0` and `eap-8.1`, provisioning `eap` should provision `eap-8.1` that would later be updated based on 8.1 channel

**Comment by jmesnil** [comment]
imho, this gives a wrong set of expectations and give the wrong impression that update is compatible across EAP minor releases which is not the case.

**Comment by TomasHofman** [comment]
> This will allow to make sure that Prospero either deal with EAP or WildFly but not both.

One question regarding Wildfly mentions (maybe you resolved it already among yourselves):

Is it at all expected that prospero would be able to provision Wildfly feature packs? I'm asking because as it was mentioned Wildfly is not going to provide channel files. Without a channel file prospero is not able to provision, because the wildfly-channel library would fail to resolve any artifact due to "no stream found":

```
$ java -jar prospero-cli/target/prospero-cli-1.0.0.Beta1-SNAPSHOT-shaded.jar install --dir wf/ --fpl 'wildfly@maven(org.jboss.universe:community-universe):current' --provision-config tmp/wildfly-config-no-channel.yaml
...
Caused by: org.wildfly.channel.UnresolvedMavenArtifactException: Can not resolve latest Maven artifact (no stream found) : org.jboss.universe:community-universe:jar:
	at org.wildfly.channel.ChannelSession.lambda$findChannelWithLatestVersion$1(ChannelSession.java:223)
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at org.wildfly.channel.ChannelSession.findChannelWithLatestVersion(ChannelSession.java:222)
	at org.wildfly.channel.ChannelSession.resolveMavenArtifact(ChannelSession.java:78)
	at org.wildfly.prospero.galleon.ChannelMavenArtifactRepositoryManager.resolve(ChannelMavenArtifactRepositoryManager.java:59)
```

In other words, IIUC,  wildfly-channel would need to change to allow resolving artifacts that are not listed in channels, or prospero would need to bypass wildfly-channel library when provisioning without channel files.

**Comment by jmesnil** [comment]
> wildfly-channel would need to change to allow resolving artifacts that are not listed in channels

It already does that. 
It seems the issue is in prospero code that does not pass the initial version of the artifact to resolve (which is used if there is no stream in the channel): 

https://github.com/wildfly-extras/prospero/blob/8b3aab3446b1925fb9d93ebbc93158e9675ddff4/prospero-common/src/main/java/org/wildfly/prospero/galleon/ChannelMavenArtifactRepositoryManager.java#L59-L60

https://github.com/wildfly-extras/prospero/blob/8b3aab3446b1925fb9d93ebbc93158e9675ddff4/prospero-common/src/main/java/org/wildfly/prospero/galleon/ChannelMavenArtifactRepositoryManager.java#L70-L71

https://github.com/wildfly-extras/prospero/blob/8b3aab3446b1925fb9d93ebbc93158e9675ddff4/prospero-common/src/main/java/org/wildfly/prospero/galleon/ChannelMavenArtifactRepositoryManager.java#L163-L164

**Comment by TomasHofman** [comment]
OK, I noticed the API change but didn't realize the consequence. Thanks :).

**Comment by TomasHofman** [comment]
@jmesnil did you mean that prospero should provide the `baseVersion` parameter to `resolveMavenArtifact()` method, or it should use different method altogether (`resolveDirectMavenArtifact`)?

It seems to me that the `baseVersion` was added to API but the value is not yet used for anything, so providing it has no effect.
